### ISSUE TITLE
Memory issue on the ffmpeg wrapper

### ISSUE THREAD

**Issue by giulioprovasi** [pascalbaljetmedia/laravel-ffmpeg#143]
There is an issue with the ffmpeg wrapper : 
```
    public function open($path): Media
    {
        $file = $this->disk->newFile($path);

        if ($this->disk->isLocal()) {
            $ffmpegPathFile = $file->getFullPath();
        } else {
            $ffmpegPathFile = static::newTemporaryFile();
            file_put_contents($ffmpegPathFile, $this->disk->read($path));
        }

        $ffmpegMedia = $this->ffmpeg->open($ffmpegPathFile);

        return new Media($file, $ffmpegMedia);
    }
```

The `$this->disk->read($path)` hit the memory limit on my server, can it be modified to `$this->disk->readStream($path)` in order to be gentler to my dear RAM ? :smile:

**Comment by pascalbaljet** [comment]
Good point, I'll take a look at it!

**Comment by pascalbaljet** [comment]
Could you try and test this branch?
https://github.com/pascalbaljetmedia/laravel-ffmpeg/tree/readStream-nonLocalFiles

I've replaced the `file_put_contents` with `stream_copy_to_stream`: https://github.com/pascalbaljetmedia/laravel-ffmpeg/commit/3a227ccafbd6bf98485216a4ed553e97ae8e5827

**Comment by giulioprovasi** [comment]
I am actually on holidays, will check the branch next week.
Cheers,
GIulio

Le mer. 7 août 2019 à 11:46, pascalbaljet <notifications@github.com> a
écrit :

**Comment by giulioprovasi** [comment]
Tested and works perfectly :)

**Comment by pascalbaljet** [comment]
Thanks for letting me know, just released v4.1.0 with this fix! :)
